android/SmartLock/app/src/main/java/org/opendds/smartlock/OpenDdsBridge.java

```diff
@@ -310,6 +310,10 @@ private void initParticipant() throws InitOpenDDSException {
             return;
         }
 
+        if (participantQos == null) {
+            throw new InitOpenDDSException("ERROR: Domain participant QOS creation failed");
```

Comment:
This exception message is misleading.  It could be that the `initParticipantFactory()` was never called.

Reply:
Right, InitParticipantFactory() is not called when a network connection is not available. So the error message should probably just say that participant QOS is not initialized.